### DISCUSSION TITLE
[Messaging] 401 null Firebase error when tries to use FCM token from different SenderId

### DISCUSSION THREAD

**Discussion by FranciscoPDNeto** [invertase/react-native-firebase#6366] [Q&A]
I'm trying to use the getToken with the SenderId option to send a firebase message from other firebase project. So basically I have the Firebase project of the app itself, where the APNs token is set, and the Firebase project of the server, where ideally would not have the app. 

But when I have this scenario, it's throwing the exception `Unexpected HTTP response with status: 401 null`, which it seems to be because of the APNs.

After I set the app on the server firebase project, with the APNs token, it started to work as expected. Even when I've deleted the app on the server firebase project, it continues to work, even for new devices (with different tokens).

Anyone knows if it's really necessary to set the app on the sender firebase project? On the [firebase documentation](https://firebase.google.com/docs/reference/ios/firebasemessaging/api/reference/Classes/FIRMessaging#-retrievefcmtokenforsenderid:completion:) it's not clear, and it sounds a little weird for me after removing the app from the sender firebase project the app continues to work

If it's not necessary, there is any other setting that I'm missing to be able to send messages through different firebase projects?

**Comment by mikehardy** [comment]
I wish I could help but this is a very particular question and I think asking the authoritative firebase repo(s) would be better for you than any speculation I/we could offer here. Could I ask you to take this to the firebase-ios-sdk repo (since it seems closer to an iOS platform question the way you have it worded)? I think you'll get more information there

**Reply by FranciscoPDNeto** [reply]
Yeah, makes sense, thanks anyway Mike!
